crates/proto/src/rr/domain/name.rs

```diff
@@ -665,13 +665,11 @@ impl Name {
         Ok(name)
     }
 
-    /// Emits the canonical version of the name to the encoder.
-    ///
-    /// In canonical form, there will be no pointers written to the encoder (i.e. no compression).
-    pub fn emit_as_canonical(
+    /// Emits the name to the encoder, with or without compression.
+    pub fn emit_with_compression(
```

Comment:
It feels like the encoder should know whether to use compression, and then `impl BinEncodable for Name` should do the right thing. Would that make sense?